### DISCUSSION TITLE
Is it possible to set value of useRef inside useAnimatedReaction?

### DISCUSSION THREAD

**Discussion by thisisgit** [software-mansion/react-native-reanimated#2835] [Q&A]
Hi, I was trying to set a value to `useRef` variable inside `useAnimatedReaction` and found out that it doesn't do anything:
```typescript
useAnimatedReaction(
  () => {
    return someBooleanValue;
  },
  (current: boolean, previous: boolean | null) => {
    if (current !== previous) {
      // current === true
      console.log('refValue before', refValue.current);  // shows `false`
      refValue.current = current;
      console.log('refValue after', refValue.current);  // still shows `false`
    }
  },
);
```
Unlike setting `useState` variable, it didn't throw any error so I was expecting this to work. 
Using `useSharedValue` instead of `useRef` works fine but I'm just curious if this is expected behavior. If it is, then I think it would be nice to display error/warning message just like when trying to set state.

**Comment by Noitidart** [comment]
I also am unsure how to use refs in reanimated. I am also getting similar undefined values.
